File: src/main/java/edu/berkeley/bid/bullet/DynamicsInfo.java
package edu.berkeley.bid.bullet;

import java.io.Serializable;

public class DynamicsInfo implements Serializable {
    private static final long serialVersionUID = 342183472184927723L;

    double m_mass;
    double [] m_localInertialPosition;
    double m_lateral_friction;

    public DynamicsInfo() {
	m_localInertialPosition = new double[3];
    }

    public DynamicsInfo(double mass, double [] localInertialPosition, double lateral_friction) {
	m_mass = mass;
	m_lateral_friction = lateral_friction;
	m_localInertialPosition = new double[3];
	m_localInertialPosition[0] = localInertialPosition[0];
	m_localInertialPosition[1] = localInertialPosition[1];
	m_localInertialPosition[2] = localInertialPosition[2];
    }

    @Override
    public String toString () {
	return "DynamicsInfo[mass=" + m_mass +
	    ",lateral_friction=" + m_lateral_friction +
	    ",localInertialPosition=[" +
	    m_localInertialPosition[0] + "," +
	    m_localInertialPosition[1] + "," +
	    m_localInertialPosition[2] + "]]";
    }

}

File: src/main/scala/BIDMach/rl/environments/bullet/DynamicsInfo.scala
package BIDMach.rl.environments.bullet;
import BIDMat.DMat;

@SerialVersionUID(473252348962354L)
class DynamicsInfo(jdi:edu.berkeley.bid.bullet.DynamicsInfo) extends Serializable {
    val javaDynamicsInfo:edu.berkeley.bid.bullet.DynamicsInfo = jdi;
    val	localInertialPosition = new DMat(1, 3, javaDynamicsInfo.m_localInertialPosition);

    def mass = javaDynamicsInfo.m_mass;
    def lateralFrictionCoeff = javaDynamicsInfo.m_lateralFrictionCoeff;

    def mass_=(v:Double) = javaDynamicsInfo.m_mass = v;
    def lateralFrictionCoeff_=(v:Double) = javaDynamicsInfo.m_lateralFrictionCoeff = v;

    def this() = this(new edu.berkeley.bid.bullet.DynamicsInfo);

    override def toString () = {
       "DynamicsInfo[mass=" + mass +
       ",lateralFrictionCoeff=" + lateralFrictionCoeff +
       ",localInertialPosition=[" + localInertialPosition(0) + "," + localInertialPosition(1) + "," + localInertialPosition(2) + "]]";
    }

}
